src/core/config/transformer/union_input_type.rs

```diff
@@ -62,6 +64,7 @@ impl<'cfg> Visitor<'cfg> {
             config,
             union_presence: HashMap::new(),
             new_types: BTreeMap::new(),
+            visited_types: Vec::new(),
```

Comment:
```suggestion
            visited_types: HashSet::new(),
```

prefer set over vec for visited checks